Make TS mechs display walk anim while turning

reaperrr opened the pull request:
Matches the original (and looks better, too).

Closes #13346.

reaperrr committed:
Make TS mechs display walk anim while turning

This matches the original, and turning without moving their legs looked silly either way.

reaperrr committed:
Add AlwaysConsiderTurnAsMove plumbing to Mobile

Can be used to make walker units like the TS mechs display move animations while turning on the spot.

Smittytron reviewed (commented):
Looks good. 
I noticed the Mammoth Mk II seems to walk while turning with or without this.

reaperrr committed:
Remove facing check from WithVoxelWalkerBody

`Mobile` now supports considering turning as `IsMoving`, so this check is no longer needed.

reaperrr commented:
> I noticed the Mammoth Mk II seems to walk while turning with or without this.

Yeah, `WithVoxelWalkerBody` did its own thing (checking both `IsMoving` *and* whether the current facing mismatches the previous facing).

I removed the now-redundant facing check from `WithVoxelWalkerBody`, please check if the Mk2 still works as intended on this PR, just to be sure.

Smittytron reviewed (dismissed):
Yep, works the same.

pchote commented on OpenRA.Mods.Common/Activities/Move/Move.cs:
```diff
@@ -225,6 +225,10 @@ public override Activity Tick(Actor self)
 				path.Add(nextCell.Value.First);
 
+				// If Mobile.Info.AlwaysConsiderTurnAsMove is true, we consider Turn as movement regardless of facing delta size.
+				// This is needed for actors that want to display their movement animation during turns (walker units, for example).
```
Can you please add to this comment where this gets set back to false?

reaperrr commented:
Updated comment.

Mailaender reviewed (approved):
Works as promised.

GraionDilach reviewed (approved):
:+1:

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/111f3e7adea7a179effb74654ac438c3e79bcfe3...c2e459f66bebda029d6de687bcbbed9b4b41eb65
